src/dev-app/dev-app/dev-app-theme.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injectable} from '@angular/core';
+import {OverlayContainer} from '@angular/cdk/overlay';
+
+const isDarkThemeKey = 'isDarkTheme';
+
+@Injectable({providedIn: 'root'})
+export class DevAppTheme {
+  readonly darkThemeClass = 'demo-unicorn-dark-theme';
+  private _elementToTheme: HTMLElement;
+  private _isDark = false;
+
+  constructor(private _overlayContainer: OverlayContainer) {
+    const isDark = localStorage.getItem(isDarkThemeKey);
+    if (isDark != null) {
+      this._isDark = Boolean(isDark);
+    }
+  }
+
+  get elementToTheme(): HTMLElement {
+    return this._elementToTheme;
+  }
+
+  set elementToTheme(value: HTMLElement) {
+    if (value != null) {
+      this._elementToTheme = value;
+      this.isDark ? this.applyDarkTheme() : this.applyLightTheme();
+    }
+  }
+
+  get isDark(): boolean {
+    return this._isDark;
+  }
+
+  set isDark(value: boolean) {
+    if (value != null) {
+      this._isDark = value;
+      localStorage.setItem(isDarkThemeKey, String(value));
+    }
+  }
```

Comment:
It feels weird that the `isDark` property handles the local storage, but the actual class updates happen only if `toggleTheme` is called. Should we just simply remove ` toggleTheme` and handle everything in the setter?

Reply:
I also handle setting the class in `set elementToTheme` since we need to wait for the element to be resolved in the component's `ngOnInit()` in order to pass it into the service and apply the class.

I'll look at applying the class to the `document.body` which could help simplify things and eliminate the need for `elementToTheme`.